Document/PostManager.php

```diff
@@ -39,6 +43,55 @@ public function findOneBySlug($year, $month, $day, $slug)
     }
 
     /**
+     * @param string        $permalink
+     * @param BlogInterface $blog
+     *
+     * @return PostInterface
+     */
+    public function findOneByPermalink($permalink, BlogInterface $blog)
```

Comment:
I used the full code of this method and added just a few lines. Don't know if it is possible to extract some stuff here.

```php
        if (isset($urlParameters['category'])) {
            $categoryQueryParts = $this->getPublicationCategoryQueryParts($urlParameters['category']);

            $parameters = array_merge($parameters, $categoryQueryParts['params']);

            $query
                ->leftJoin('p.category', 'cat')
                ->andWhere($categoryQueryParts['query']);
        }
```